EFCore.CheckConstraints/Internal/ValidationCheckConstraintConvention.cs

```diff
@@ -0,0 +1,260 @@
+using Microsoft.EntityFrameworkCore.Storage;
+using System.Reflection;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata;
+using Microsoft.EntityFrameworkCore.Metadata.Conventions;
+
+namespace EFCore.CheckConstraints.Internal
+{
+    /// <summary>
+    ///     A convention that creates check constraints for various validation attributes.
+    /// </summary>
+    public class ValidationCheckConstraintConvention : IModelFinalizingConvention
+    {
+        public const string DefaultPhoneRegex = @"^[\d\s+-.()]*\d[\d\s+-.()]*((ext\.|ext|x)\s*\d+)?\s*$";
+        public const string DefaultCreditCardRegex = @"^[\d- ]*$";
+        public const string DefaultEmailAddressRegex = @"^[^@]+@[^@]+$";
+        public const string DefaultUrlAddressRegex = @"^(http://|https://|ftp://)";
+
+        private readonly ISqlGenerationHelper _sqlGenerationHelper;
+        private readonly IDatabaseProvider _databaseProvider;
+        private readonly RelationalTypeMapping _intTypeMapping;
+
+        private readonly bool _supportsRegex;
+        private readonly string _phoneRegex, _creditCardRegex, _emailAddressRegex, _urlRegex;
+
+        public ValidationCheckConstraintConvention(
+            ValidationCheckConstraintOptions options,
+            ISqlGenerationHelper sqlGenerationHelper,
+            IRelationalTypeMappingSource relationalTypeMappingSource,
+            IDatabaseProvider databaseProvider)
+        {
+            _sqlGenerationHelper = sqlGenerationHelper;
+            _databaseProvider = databaseProvider;
+            _intTypeMapping = relationalTypeMappingSource.FindMapping(typeof(int));
+
+            _supportsRegex = SupportsRegex;
+            _phoneRegex = options.PhoneRegex ?? DefaultPhoneRegex;
+            _creditCardRegex = options.CreditCardRegex ?? DefaultCreditCardRegex;
+            _emailAddressRegex = options.EmailAddressRegex ?? DefaultEmailAddressRegex;
+            _urlRegex = options.UrlRegex ?? DefaultUrlAddressRegex;
+        }
+
+        /// <inheritdoc />
+        public virtual void ProcessModelFinalizing(IConventionModelBuilder modelBuilder, IConventionContext<IConventionModelBuilder> context)
+        {
+            var sql = new StringBuilder();
+
+            foreach (var entityType in modelBuilder.Metadata.GetEntityTypes())
+            {
+                var tableName = entityType.GetTableName();
+                if (tableName is null)
+                {
+                    continue;
+                }
+
+                foreach (var property in entityType.GetDeclaredProperties().Where(p => p.PropertyInfo != null || p.FieldInfo != null))
+                {
+                    var memberInfo = (MemberInfo)property.PropertyInfo ?? property.FieldInfo;
+                    if (memberInfo is null)
+                    {
+                        continue;
+                    }
+
+                    var columnName = property.GetColumnName();
```

Comment:
Should get the name for the specific table

Reply:
Thanks, still learning the new world of the relational model. Will also modify the other conventions to do the same.

Should the parameterless GetColumnName continue to exist, as it's rarely what you want to call? Maybe rename to GetDefaultColumnName (the breaking change here would probably have been a good thing)?

Reply:
Neither should be used outside of model building and I think there's not enough benefit to do this just for conventions.

Reply:
FWIW in EFCore.PG there are calls to parameterless GetColumnName in [NpgsqlMigrationsSqlGenerator](https://github.com/npgsql/efcore.pg/blob/210fc07925be63545fd46922766d185b9ced51ad/src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs#L67) and in [NpgsqlAnnotationProvider](https://github.com/npgsql/efcore.pg/blob/ac9c7d2b8c56ac465e56f2fac2806343f08d3d20/src/EFCore.PG/Metadata/Internal/NpgsqlAnnotationProvider.cs#L77), which I will now go and fix.

I'm pretty sure that any provider writer out there will call this wrong method 100%, it seems dangerous.

Reply:
It's too late for 5.0 anyway. I would recommend provider writers to add TPT tests for provider-specific functionality.